worlds/wargroove2/__init__.py

```diff
@@ -0,0 +1,194 @@
+from random import Random
+
+import settings
+import string
+import typing
+
+from BaseClasses import Item, Tutorial, ItemClassification
+from Options import NumericOption
+from .Items import item_table, faction_table, Wargroove2Item
+from .Levels import Wargroove2Level, get_level_table, get_first_level, get_final_levels, region_names, FINAL_LEVEL_1, \
+    FINAL_LEVEL_2, FINAL_LEVEL_3, FINAL_LEVEL_4, LEVEL_COUNT, FINAL_LEVEL_COUNT
+from .Locations import location_table
+from .Presets import wargroove2_option_presets
+from .Regions import create_regions
+from .Rules import set_rules
+from worlds.AutoWorld import World, WebWorld
+from .Options import Wargroove2Options
+from worlds.LauncherComponents import Component, components, Type, launch_subprocess
+
+
+def launch_client():
+    from .client import launch
+    launch_subprocess(launch, name="Wargroove2Client")
+
+
+components.append(Component("Wargroove 2 Client", "Wargroove2Client", func=launch_client, component_type=Type.CLIENT))
+
+
+class Wargroove2Settings(settings.Group):
+    class RootDirectory(settings.UserFolderPath):
+        """
+        Locate the Wargroove 2 root directory on your system.
+        This is used by the Wargroove 2 client, so it knows where to send communication files to
+        """
+        description = "Wargroove 2 root directory"
+
+    root_directory: RootDirectory = RootDirectory("C:/Program Files (x86)/Steam/steamapps/common/Wargroove 2")
+
+
+class Wargroove2Web(WebWorld):
+    tutorials = [Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up Wargroove 2 for Archipelago.",
+        "English",
+        "wargroove2_en.md",
+        "wargroove2/en",
+        ["Fly Sniper"]
+    )]
+
+    options_presets = wargroove2_option_presets
+
+
+class Wargroove2World(World):
+    """
+    Command an army, in the sequel to the hit turn based strategy game Wargroove!
+    """
+    options: Wargroove2Options
+    options_dataclass = Wargroove2Options
+    settings: typing.ClassVar[Wargroove2Settings]
+    game = "Wargroove 2"
+    topology_present = True
+    web = Wargroove2Web()
+    level_list: [Wargroove2Level] = None
+    first_level: Wargroove2Level = None
+    final_levels: [Wargroove2Level] = None
```

Comment:
So the typing here would technically be this:
```suggestion
    level_list: typing.Optional[typing.List[Wargroove2Level]] = None
    first_level: typing.Optional[Wargroove2Level] = None
    final_levels: typing.Optional[typing.List[Wargroove2Level]] = None
```
But this breaks a lot of assumptions in your code since I don't see any reason you ever want these to be `None`. I think what you want here is just
```suggestion
    level_list: typing.List[Wargroove2Level] = []
    first_level: Wargroove2Level
    final_levels: typing.List[Wargroove2Level] = []
```